Subject: Provide boot option to disable sched_rt_runtime
From: Jack Steiner <steiner@sgi.com>
Patch-mainline: no
References: bnc#610783

Very large machines may livelock due to scalability issues, so give the
user the ability to boot time disable realtime bandwidth control.

Signed-off-by: Mike Galbraith <mgalbraith@suse.de>

---
 Documentation/kernel-parameters.txt |    2 ++
 kernel/sched.c                      |   11 +++++++++++
 2 files changed, 13 insertions(+)

--- a/Documentation/kernel-parameters.txt
+++ b/Documentation/kernel-parameters.txt
@@ -2261,6 +2261,8 @@ bytes respectively. Such letter suffixes
 
 	sched_debug	[KNL] Enables verbose scheduler debug messages.
 
+	nortsched       [KNL] Disables realtime task bandwidth control.
+
 	security=	[SECURITY] Choose a security module to enable at boot.
 			If this boot parameter is not specified, only the first
 			security module asking for security registration will be
--- a/kernel/sched.c
+++ b/kernel/sched.c
@@ -941,6 +941,17 @@ static __read_mostly int scheduler_runni
  */
 int sysctl_sched_rt_runtime = 950000;
 
+/*
+ * Bootline option to disable sched_rt_runtime.  Otherwise very large
+ * systems livelock during boot.
+ */
+static int __init parse_nortsched(char *arg)
+{
+	sysctl_sched_rt_runtime = -1;
+	return 0;
+}
+early_param("nortsched", parse_nortsched);
+
 static inline u64 global_rt_period(void)
 {
 	return (u64)sysctl_sched_rt_period * NSEC_PER_USEC;
